### PR TITLE
Add ubuntu 22.04 (jammy) support

### PR DESCRIPTION
**Pull request by jforest**
So i can create a new set of r-base images based on jammy, so I can build an updated revdepcheck-cloud docker image based on those 🎪

### CODE REVIEW

File: builder/Dockerfile.ubuntu-2204 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  && sed -i "s|# deb-src|deb-src|g" /etc/apt/sources.list \
+  && export DEBIAN_FRONTEND=noninteractive \
+  && apt-get update \
+  && apt-get install -y libblas-dev libcurl4-openssl-dev libicu-dev liblapack-dev libpcre2-dev wget python3-pip ruby ruby-dev \
+  && apt-get build-dep -y r-base
+
+RUN pip3 install awscli
+
+# Pin fpm to avoid git dependency in 1.12.0
+RUN gem install fpm:1.11.0
```
Comment (glin) [inline builder/Dockerfile.ubuntu-2204:15]: I ran into an issue with pinning fpm to 1.11.0 for Debian 11 (https://github.com/rstudio/r-builds/pull/106), and see the same error on Ubuntu 22:
```sh
...
Traceback (most recent call last):
	9: from /usr/local/bin/fpm:25:in `<main>'
	8: from /usr/local/bin/fpm:25:in `load'
	7: from /var/lib/gems/2.7.0/gems/fpm-1.11.0/bin/fpm:7:in `<top (required)>'
	6: from /var/lib/gems/2.7.0/gems/clamp-1.0.1/lib/clamp/command.rb:133:in `run'
	5: from /var/lib/gems/2.7.0/gems/fpm-1.11.0/lib/fpm/command.rb:574:in `run'
	4: from /var/lib/gems/2.7.0/gems/clamp-1.0.1/lib/clamp/command.rb:68:in `run'
	3: from /var/lib/gems/2.7.0/gems/fpm-1.11.0/lib/fpm/command.rb:487:in `execute'
	2: from /var/lib/gems/2.7.0/gems/fpm-1.11.0/lib/fpm/package/deb.rb:494:in `output'
	1: from /var/lib/gems/2.7.0/gems/fpm-1.11.0/lib/fpm/package/deb.rb:494:in `tap'
/var/lib/gems/2.7.0/gems/fpm-1.11.0/lib/fpm/package/deb.rb:495:in `block in output': uninitialized constant FPM::Package::Deb::Zlib (NameError)
```
I ended up installing the latest (1.13.1 at the time) it to get it working, but it's kind of a headache either way. Both pinned and latest installs have caused issues in the past.

**Review by glin — COMMENTED** [review]
Looks good with a couple minor comments.

We'll need to add Ubuntu 22 to the skipped builds for R 3.3.x as well: https://github.com/rstudio/r-builds/blob/61a629916852cd066906c4caeaa1cba5748c601b/handler.py#L127-L138

And Ubuntu 22 might have this GCC 10+ issue when compiling for R 3.6.0 and below, that I ran into with the Debian 11 builds: https://github.com/rstudio/r-builds/pull/106. If the R <= 3.6.0 builds end up failing, I can merge https://github.com/rstudio/r-builds/pull/106 to get a general fix for any distros that ship with GCC 10+.

**Comment by jforest** [comment]
the checks have failed due to centos8, which is not anything I have touched, so I think those can be ignored.

```
[](https://build.rstudioservices.com/blue/organizations/jenkins/r-builds%2Fr-builds/detail/add-ubuntu-2204/2/pipeline#step-14-log-590)[](https://build.rstudioservices.com/blue/organizations/jenkins/r-builds%2Fr-builds/detail/add-ubuntu-2204/2/pipeline#step-14-log-591)[](https://build.rstudioservices.com/blue/organizations/jenkins/r-builds%2Fr-builds/detail/add-ubuntu-2204/2/pipeline#step-14-log-592)[2022-02-04T13:43:43.021Z] ERROR: for centos-8  (<Service: centos-8>, 'Build failed')

[2022-02-04T13:43:43.021Z] Service 'centos-8' failed to build : Build failed

[2022-02-04T13:43:43.021Z] make: *** [Makefile:8: docker-build] Error 1

script returned exit code 2
```

File: serverless-resources.yml [hunk 2]
```diff
@@ -173,6 +173,20 @@ rBuildsBatchJobDefinitionUbuntu2004:
       Image: "#{AWS::AccountId}.dkr.ecr.#{AWS::Region}.amazonaws.com/r-builds:ubuntu-2004"
     Timeout:
       AttemptDurationSeconds: 7200
+rBuildsBatchJobDefinitionUbuntu2204:
```
Comment (glin) [inline serverless-resources.yml:176]: Oh yeah, this reminds me, we also need to update serverless.yml and the builder/docker-compose.yml file.

https://github.com/rstudio/r-builds/blob/61a629916852cd066906c4caeaa1cba5748c601b/serverless.yml#L34-L60

And then I think we're good. I'm working on related CentOS 8 issues now, and can look at migrating to Rocky here.

**Review by glin — APPROVED** [review]

**Pull request merged**